### DISCUSSION TITLE
ERROR: No 'composer.lock' found (Heroku Deployment Error)

### DISCUSSION THREAD

**Discussion by muratumutlu** [directus/directus#2696] [Archive]
Hi guys,

I'm trying Directus for 20 days. It is really scalable and awesome for React / React native apps.

Today, I tried to deploy directus to heroku.
1) Forked directus project.
2) Gave Heroku to access forked project with Github

But I got an error like these:

`ERROR: No 'composer.lock' found!
 !     
 !     A 'composer.lock' file was not found in your project, but there
 !     is a 'composer.json' file with dependencies inside 'require'.
 !     
 !     The lock file is required in order to guarantee reliable and
 !     reproducible installation of dependencies across platforms and
 !     deploys. You must follow the Composer best practice of having
 !     your lock file under version control in order to deploy. The
 !     lock file must not be in your '.gitignore'.
 !     
 !     Please perform the following steps locally on your computer to
 !     resolve this issue before attempting another deploy:
 !     1) remove 'composer.lock' from file '.gitignore', if present
 !     2) if no 'composer.lock' exists, run 'composer update'
 !     3) stage the lock file changes using 'git add composer.lock'
 !     4) if you edited '.gitignore', also run 'git add .gitignore'
 !     5) commit the change using 'git commit'
 !     
 !     Please remember to always keep your 'composer.lock' updated in
 !     lockstep with 'composer.json' to avoid common problems related
 !     to dependencies during collaboration and deployment.
 !     
 !     Please refer to the Composer documentation for further details:
 !     https://getcomposer.org/doc/
 !     https://getcomposer.org/doc/01-basic-usage.md
 !     Push rejected, failed to compile PHP app.
 !     Push failed`

**Comment by rijkvanzanten** [comment]
(man that formatting with the `!` is annoying to read)

@muratumutlu Feel free to add it in in a PR 🙂  (and resolve https://github.com/directus/api/issues/1845 in the process)
